src/Chapter4.hs

```diff
@@ -610,7 +634,17 @@ Implement the 'Monad' instance for our lists.
 🕯 HINT: You probably will need to implement a helper function (or
   maybe a few) to flatten lists of lists to a single list.
 -}
+instance Monad List where
+    (>>=) :: List a -> (a -> List b) -> List b
+    l >>= f = flattenList (fmap f l)
 
+flattenList :: List (List a) -> List a
+flattenList Empty       = Empty
+flattenList (Cons x xs) = concatenateList x (flattenList xs)
+
+concatenateList :: List a -> List a -> List a
+concatenateList Empty x        = x
```

Comment:
Perfect!

You can one more case to make your solution even more efficient:

```haskell
concatenateList xs Empty = xs
```

In that case, you won't need to walk through each element of the first list when the second one is empty and return the list instantly 🚤